### PR TITLE
MDS-3308: Users Can Add Author, Title, and Date Info For Final Application Package Documents & Previous Amendment Documents (If Applicable)

### PR DESCRIPTION
**Pull request by exzizt**
# Main
* Users can edit metadata (title, author, date) for documents in the Final Application Package and for documents related to the previous permit (if applicable) in the Draft Permit tab while processing a permit. Title is mandatory while author and date are not.

## Screenshots
image

### CODE REVIEW

File: migrations/sql/V2021.01.15.11.12__update_now_application_document_xref.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
+ALTER TABLE now_application_document_xref
+ADD COLUMN IF NOT EXISTS preamble_title varchar,
+ADD COLUMN IF NOT EXISTS preamble_author varchar,
+ADD COLUMN IF NOT EXISTS preamble_date timestamptz;
+
+ALTER TABLE now_application_document_identity_xref
```
Comment (exzizt) [inline migrations/sql/V2021.01.15.11.12__update_now_application_document_xref.sql:6]: For imported NoW submission documents that are part of the FAP.

File: migrations/sql/V2021.01.15.11.12__update_now_application_document_xref.sql [hunk 2]
```diff
@@ -0,0 +1,14 @@
+ALTER TABLE now_application_document_xref
```
Comment (exzizt) [inline migrations/sql/V2021.01.15.11.12__update_now_application_document_xref.sql:1]: For requested NoW documents that are part of the FAP.

Comment (exzizt) [inline migrations/sql/V2021.01.15.11.12__update_now_application_document_xref.sql:1]: Thinking of renaming these to remove "preamble"?

**Review by nayr974 — DISMISSED** [review]
image

**Review by nayr974 — APPROVED** [review]

**Pull request merged**